src/vm/core.zig

```diff
@@ -781,3 +797,84 @@ test "update fp dst felt" {
     const fp = vm.getFp();
     try expectEqual(fp.offset, 42);
 }
+
+test "trace is enabled" {
+    // ************************************************************
+    // *                 SETUP TEST CONTEXT                       *
+    // ************************************************************
+    // Initialize an allocator.
+    var allocator = std.testing.allocator;
+
+    // Create a new VM instance.
+    var vm = try CairoVM.init(&allocator, true);
+    defer vm.deinit();
+
+    // ************************************************************
+    // *                      TEST BODY                           *
+    // ************************************************************
+    // Do nothing
+
+    // ************************************************************
+    // *                      TEST CHECKS                         *
+    // ************************************************************
+    // Check that trace was initialized
+    const trace = vm.trace;
+
+    if (trace == null) {
+        return error.TraceShouldHaveBeenEnabled;
+    }
+}
+
+test "trace is disabled" {
+    // ************************************************************
+    // *                 SETUP TEST CONTEXT                       *
+    // ************************************************************
+    // Initialize an allocator.
+    var allocator = std.testing.allocator;
+
+    // Create a new VM instance.
+    var vm = try CairoVM.init(&allocator, false);
+    defer vm.deinit();
+
+    // ************************************************************
+    // *                      TEST BODY                           *
+    // ************************************************************
+    // Do nothing
+
+    // ************************************************************
+    // *                      TEST CHECKS                         *
+    // ************************************************************
+    // Check that trace was initialized
+    const trace = vm.trace;
+
+    if (trace != null) {
+        return error.TraceShouldHaveBeenDisabled;
+    }
+}
+
+// TODO: Find a valid Instruction so we can call `runInstruction` to check if traces
```

Comment:
remove commented test and create an issue with the content of this test instead please